age/keysource.go

```diff
@@ -169,55 +210,84 @@ func (key *MasterKey) ToString() string {
 
 // ToMap converts the MasterKey to a map for serialization purposes.
 func (key *MasterKey) ToMap() map[string]interface{} {
-	return map[string]interface{}{"recipient": key.Recipient, "enc": key.EncryptedKey}
+	out := make(map[string]interface{})
+	out["recipient"] = key.Recipient
+	out["enc"] = key.EncryptedKey
+	return out
 }
 
-// MasterKeysFromRecipients takes a comma-separated list of Bech32-encoded public keys and returns a
-// slice of new MasterKeys.
-func MasterKeysFromRecipients(commaSeparatedRecipients string) ([]*MasterKey, error) {
-	if commaSeparatedRecipients == "" {
-		// otherwise Split returns [""] and MasterKeyFromRecipient is unhappy
-		return make([]*MasterKey, 0), nil
-	}
-	recipients := strings.Split(commaSeparatedRecipients, ",")
-
-	var keys []*MasterKey
+// loadIdentities attempts to load the age identities based on runtime
+// environment configurations (e.g. SopsAgeKeyEnv, SopsAgeKeyFileEnv,
+// SopsAgeKeyUserConfigPath). It will load all found references, and expects
+// at least one configuration to be present.
+func (key *MasterKey) loadIdentities() (ParsedIdentities, error) {
+	var readers = make(map[string]io.Reader, 0)
 
-	for _, recipient := range recipients {
-		key, err := masterKeyFromRecipient(recipient)
+	if ageKey, ok := os.LookupEnv(SopsAgeKeyEnv); ok {
+		readers[SopsAgeKeyEnv] = strings.NewReader(ageKey)
+	}
 
+	if ageKeyFile, ok := os.LookupEnv(SopsAgeKeyFileEnv); ok {
+		f, err := os.Open(ageKeyFile)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to open %s file: %w", SopsAgeKeyFileEnv, err)
```

Comment:
The other option would be to log any loading failure, and only error if `identities == 0`.

Reply:
I think this way is preferable. A bit more defense around user mistakes, especially in automation using sops.